stringdtype/stringdtype/src/dtype.c

```diff
@@ -190,10 +229,22 @@ nonzero(void *data, void *NPY_UNUSED(arr))
 int
 compare(void *a, void *b, void *NPY_UNUSED(arr))
 {
-    ss *ss_a = NULL;
-    ss *ss_b = NULL;
-    load_string(a, &ss_a);
-    load_string(b, &ss_b);
+    ss *ss_a = (ss *)a;
+    ss *ss_b = (ss *)b;
+    int a_is_null = ss_isnull(ss_a);
+    int b_is_null = ss_isnull(ss_b);
+    if (a_is_null || b_is_null) {
+        // numpy sorts NaNs to the end of the array
+        // pandas sorts NAs to the end as well
+        // so we follow that behavior here
+        if (!b_is_null) {
+            return 1;
+        }
+        else if (!a_is_null) {
+            return -1;
+        }
+        return 0;
+    }
```

Comment:
I think this can this be simplified, since if either a or b is null we don't need to test the other. 

```c
if (a_is_null) {
    return 1;
}
if (b_is_null) {
    return -1;
}
return strcmp(ss_a->buf, ss_b->buf);
```